GitHub Actions doesn't work, cmake doesn't build with conan

Hi,
I forked the repo and tried to build it with conan on my Mac, but faced some difficulties. I'm new to C++, it would be great if you could help.
1. I try to run GitHub Action in my forked repo, but it fails
my run is [here](https://github.com/LucienZhang/archicad-addon-cmake/actions/runs/4243549735/jobs/7376442936), the only change I made is [here](https://github.com/GRAPHISOFT/archicad-addon-cmake/compare/master...LucienZhang:archicad-addon-cmake:master)
2. I tried to build the project with conan, but seems cmake failed reading `AC_API_DEVKIT_DIR`.

My environment
```
MacOS=13.2.1
Python=3.10.8
cmake version 3.25.2
Conan version 1.59.0
```

Log
```sh
$ conan install . -pr:b=default --install-folder=Build/
Configuration (profile_host):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13
os=Macos
os_build=Macos
[options]
[build_requires]
[env]

archicad-apidevkit/26: Main binary package '4b92821f1af51ada4198018b78c8b197b32c6c18' missing. Using compatible package '2cd20192e71f56e1115fbdc2ebf2871bb61152e4'
conanfile.py: Installing package
Requirements
    archicad-apidevkit/26 from 'conancenter' - Cache
Packages
    archicad-apidevkit/26:2cd20192e71f56e1115fbdc2ebf2871bb61152e4 - Cache

Installing (downloading, building) binaries...
archicad-apidevkit/26: Already installed!
archicad-apidevkit/26: Setting AC_API_DEVKIT_DIR environment variable: /Users/lucien/.conan/data/archicad-apidevkit/26/_/_/package/2cd20192e71f56e1115fbdc2ebf2871bb61152e4/bin
conanfile.py: Generator 'CMakeToolchain' calling 'generate()'
conanfile.py: Preset 'release' added to CMakePresets.json. Invoke it manually using 'cmake --preset release'
conanfile.py: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE=/Users/lucien/Projects/compliance_helper/conan/conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
conanfile.py: Generator 'CMakeDeps' calling 'generate()'
conanfile.py: Generator cmake_paths created conan_paths.cmake
conanfile.py: Generator txt created conanbuildinfo.txt
conanfile.py: Generator 'VirtualBuildEnv' calling 'generate()'
conanfile.py: Aggregating env generators
conanfile.py: Generated conaninfo.txt
conanfile.py: Generated graphinfo
```

```sh
$ cd Build && cmake -G "Xcode" ..
CMake Error at CMakeLists.txt:57 (message):
  Failed to detect Archicad version, please check the value of the
  AC_API_DEVKIT_DIR variable.


-- Configuring incomplete, errors occurred!
```

```sh
$ grep "export AC_API_DEVKIT_DIR" ./conan/conanbuildenv-release-x86_64.sh
export AC_API_DEVKIT_DIR="/Users/lucien/.conan/data/archicad-apidevkit/26/_/_/package/2cd20192e71f56e1115fbdc2ebf2871bb61152e4/bin"

$ grep "AC_API_DEVKIT_DIR" ./Build/CMakeCache.txt
AC_API_DEVKIT_DIR:PATH=
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -22,7 +22,9 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Get Conan
-      uses: turtlebrowser/get-conan@v1.0
+      uses: turtlebrowser/get-conan@v1.2
+      with:
+        version: 1.59.0
       
     - name: Create Conan default profile
       run: conan profile new default --detect

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,7 +32,7 @@ endfunction ()
 
 set (API_DEVKIT_DIR $ENV{AC_API_DEVKIT_DIR})
 
-if (GITHUB_BUILD)
+if (NOT EXISTS ${API_DEVKIT_DIR})
 	include(${CMAKE_BINARY_DIR}/conan_paths.cmake)
 	set (API_DEVKIT_DIR ${CONAN_ARCHICAD-APIDEVKIT_ROOT}/bin)
 endif ()

--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@ This repository contains a CMake template for Archicad Add-On Development. You c
 
 - [CMake](https://cmake.org) (3.16 minimum version is needed).
 - [Python](https://www.python.org) for resource compilation (version 2.7+ or 3.8+).
-- [Conan](https://conan.io) for convenience.
+- [Conan](https://conan.io) for convenience (version 1.x).
 
 There are several possibilities to build this Add-On, using different IDEs, with downloaded Archicad API Development Kit or by using the Conan C++ package manager.
 
@@ -21,7 +21,7 @@ There are several possibilities to build this Add-On, using different IDEs, with
   - `AC_ADDON_NAME`: (optional) The name of the project file and the result binary Add-On file (default is "ExampleAddOn").
   - `AC_ADDON_LANGUAGE`: (optional) The language code of the Add-On (default is "INT").
   - `AC_MDID_DEV`: (optional) Your Developer ID. Omitting this will result in a 1 value.
-  - `AC_MDID_LOC`: (optional) Add-On Local ID. Omitting this will result in a 1 value. 
+  - `AC_MDID_LOC`: (optional) Add-On Local ID. Omitting this will result in a 1 value.
 - To release your Add-On you have to provide valid MDIDs.
 
 ### Visual Studio (Windows)
@@ -54,7 +54,7 @@ cd ..
 
 ## Build using the Conan Package Manager
 ### Prepare
-- [Download the package manager](https://conan.io/downloads.html)
+- [Download the package manager](https://conan.io/downloads.html), please install conan v1.x.
 - Run conan to create the default profile (use Command Prompt on Windows and Terminal on MacOS):
 
       conan profile new default --detect
@@ -95,11 +95,11 @@ cd ..
 - Open the root folder in Visual Studio Code, configure and build the solution.
 ## Archicad Compatibility
 
-This template is tested with all Archicad versions starting from Archicad 23 using the downloaded Archicad API Development Kit and starting fro Archicad 25 using Conan. 
+This template is tested with all Archicad versions starting from Archicad 23 using the downloaded Archicad API Development Kit and starting fro Archicad 25 using Conan.
 
 ## Use in Archicad
 
-To use the Add-On in Archicad, you have to add your compiled .apx file in Add-On Manager. The example Add-On registers a new command into the Options menu. Please note that the example Add-On works only in the demo version of Archicad. 
+To use the Add-On in Archicad, you have to add your compiled .apx file in Add-On Manager. The example Add-On registers a new command into the Options menu. Please note that the example Add-On works only in the demo version of Archicad.
 
 You can start Archicad in demo mode with the following command line commands:
 - Windows: `ARCHICAD.exe -DEMO`

